warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryPruningVisitor.java

```diff
@@ -576,6 +614,21 @@ private void replaceAndAssign(JexlNode toReplace, String queryString, JexlNode b
                     JexlNodes.children(parent, children.toArray(new JexlNode[children.size()]));
                 }
             }
+            
+            // if a junction was completely pruned it is replaced with the last child. In this case, also check
+            // for extra reference expressions and prune those.
+            if ((toReplace instanceof ASTOrNode || toReplace instanceof ASTAndNode) && (parent != null && parent.jjtGetNumChildren() == 1)) {
+                if (parent instanceof ASTReferenceExpression && parent.jjtGetParent() != null) {
+                    JexlNode grandparent = parent.jjtGetParent();
+                    if (grandparent instanceof ASTReference && grandparent.jjtGetParent() != null) {
+                        JexlNode greatGrandParent = grandparent.jjtGetParent();
+                        JexlNodes.swap(greatGrandParent, grandparent, parent.jjtGetChild(0));
+                        
+                        // blank out the parent and grandparent
+                        JexlNodes.blank(parent, grandparent);
```

Comment:
Is this blanking really necessary?  The blank method creates an extra, unused JexlNode which eventually is just going to be garbage collected anyway.

Reply:
The motivation here is to eliminate pointers from nodes that are pruned from the tree. Just because a node is pruned, it may still have references to valid nodes in the resulting query tree, and as the visitor pops back up the call stack, it will visit the pruned nodes. 

In the case that the visitor tries to do any secondary operations on a node after the initial visit, this blanking would prevent any accidental secondary visits.

This is defensive in nature, not because of any known bug.